CMakeLists.txt

```diff
@@ -72,6 +72,10 @@ add_compile_options(-Wvla)
 add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-fpermissive>)
 add_compile_options(-Wno-unused-function -Wno-deprecated-declarations -Wno-unknown-pragmas)
 
+if (USING_CLANG)
+  add_compile_options(-Wthread-safety -Wthread-safety-negative)
```

Comment:
ah yes more warning on Windows 😟

Reply:
let me guess: you're going to end up converting the _entire_ codebase over to abseil at some point ~~runs~~

Reply:
are we using clang on windows? We can tweak this more if you want?

I'd prefer to keep with std where possible. abseil synchronisation primitives have some useful features. 

std > abseil >> writing it ourselves

Reply:
>clang on windows

@michael-loki yes, in fact, i've spent the weekend cutting over to new solaris cross-compile box, only to find that clang v8 RC3 is still a bit broken: https://twitter.com/__rvx86/status/1101709699127349249

Reply:
even worse, this is *before* the abseil integration, and in ***debug*** mode: https://spit.mixtape.moe/view/39b1a1db#WSOIKsFTA1vtBm2vmwiW0oGv7AtODkiC

Reply:
yeah that's weird AF.

```c++
$ echo __ZNKSt9type_infoeqERKS_ | c++filt
std::type_info::operator==(std::type_info const&) const
```

`ccache` is magic.